### DISCUSSION TITLE
Modeling `Decimal` types as json-schema results in ambiguous typing

### DISCUSSION THREAD

**Discussion by bruno-f-cruz** [AllenNeuralDynamics/aind-data-schema#739] [Ideas]
**Describe the bug**
The json model builder for `Decimal` types results in the following signature:
```python
"anyOf": [
            {
              "type": "number"
            },
            {
              "type": "string"
            }
          ],
```
This was a change from pydantic v1 that allows decimal numbers to be encoded as strings in order to preserve their numeric precision. ([see details here](https://docs.pydantic.dev/latest/api/standard_library_types/#decimaldecimal)).

Unfortunately, Pydantic goes a bit rogue and comes up with their own standard (at least I couldn't find anything in json-schema nor OpenApi??) of what a Decimal means by choosing a AnyOf[str | number] and solving the deserialization with custom logic ([see source here](https://github.com/pydantic/pydantic/blob/8060fa1cff965850e5e08a67ca73d5272dcdcf9f/pydantic/json_schema.py#L615)).

This is unfortunate since the union is not discriminated, and thus makes it impossible to know what the type should resolve as.

Other users have also raised this issue. [See this discussion for more details](https://github.com/pydantic/pydantic/issues/7457#issuecomment-1731314889)

**Describe the solution you'd like**
It would be nice to change the model class and force decimals to either always be strings or numbers. This can be done in two ways:
1. Set the property of the model_json() method to `Serialize` instead of the default `Validate` 
2.  Override the `decimal_schema` method in the implementation mentioned above and create our own custom model builder.

**Comment by dyf** [comment]
I agree that this is counterintuitive. I haven't thought too hard about it, but I wonder if we should just pivot and go with `float` everywhere instead of `Decimal`.  This comes at a cost of precision issues, but I'm not sure how much of a problem that is. Thoughts?

@bruno-f-cruz @jtyoung84 @saskiad

**Reply by bruno-f-cruz** [reply]
I am happy to go with float.
